### ISSUE TITLE
What is the initial hidden state of LSTM/GRU during training?

### ISSUE THREAD

**Issue by Haitons** [Websail-NU/torch-defseq#4]

**Comment by northanapon** [comment]
It's set to zero.

**Comment by Haitons** [comment]
So how to use seed information for LSTM？

**Comment by northanapon** [comment]
You can start the sentence with `w <def>`, where `w` is the word being defined.

**Comment by Haitons** [comment]
Thanks. And LSTM has two hidden states `(h,c)`,when use `H` or  `G` to update the hidden state,do we both change or just change one of them?

**Comment by northanapon** [comment]
I am not sure what you were referring to `H` or `G`. An LSTM cell will update both `h` and `g`, but our gate only modifies `h` afterward (which does not change the LSTM cell's state in the next time step).

**Comment by Haitons** [comment]
I still don't quite understand the meaning of the second sentence......
`H` refers to the model `Hidden`，`G` refers to the model `Gated`.I use pytorch nn.LSTM,in every time step,It outputs LSTM hidden state contains `（h，c）`，we get it and use `Hidden` or `Gated` to update it，I am not sure whether to update h or c, or update them all......

**Comment by northanapon** [comment]
Oh, now I understand. In both `Hidden` and `Gated` model, we only take `h` from the LSTM cell and then "combine" `h` with the word being defined features `v*`.

**Comment by Haitons** [comment]
Yeah!I have got it ,thank you my friend.
